### PR TITLE
[#11132] fix(iceberg-rest): Fix NamespaceNotEmptyException to return HTTP 409

### LINKED ISSUE #11132: [Subtask] Fix NamespaceNotEmptyException to return HTTP 409 instead of 400
## Describe the subtask

The Iceberg REST spec mandates HTTP 409 Conflict for `dropNamespace` when the namespace is non-empty. The current mapping in `IcebergExceptionMapper.java` returns 400, and the test at `TestIcebergExceptionMapper.java:51` asserts the wrong status code.

**Files:**
- `iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/IcebergExceptionMapper.java`
- `iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestIcebergExceptionMapper.java:51`
- `iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java:224`

**Fix:** Update the mapping to return 409 for `NamespaceNotEmptyException` and correct the test assertion from 400 to 409.

## Parent issue

https://github.com/apache/gravitino/issues/11131

### PR DESCRIPTION
**Pull request by lasdf1234**
### What changes were proposed in this pull request?

Map `NamespaceNotEmptyException` to HTTP 409 Conflict in `IcebergExceptionMapper`, aligned with the Iceberg REST spec and upstream `RESTCatalogAdapter`. Stop converting `NamespaceNotEmptyException` to `BadRequestException` in `convertToIcebergException`. Update unit tests and add REST test coverage for dropping a non-empty namespace.

### Why are the changes needed?

The Iceberg REST spec requires HTTP 409 Conflict when `dropNamespace` is called on a non-empty namespace. Gravitino currently returns 400 Bad Request, which is non-conformant.

Fix: #11132

### Does this PR introduce _any_ user-facing change?

Yes. Clients receive HTTP 409 Conflict instead of 400 Bad Request when dropping a non-empty namespace. The error type remains `NamespaceNotEmptyException`.

### How was this patch tested?

- `./gradlew :iceberg:iceberg-rest-server:test --tests "org.apache.gravitino.iceberg.service.TestIcebergExceptionMapper" -PskipITs`
- `./gradlew :iceberg:iceberg-rest-server:test --tests "org.apache.gravitino.iceberg.service.rest.TestIcebergNamespaceOperations#testDropNamespace" -PskipITs`
- `./gradlew :iceberg:iceberg-rest-server:test --tests "org.apache.gravitino.iceberg.integration.test.IcebergRESTMemoryCatalogIT#testDropNameSpace" -PskipITs`

### CODE REVIEW

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestIcebergExceptionMapper.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ private void checkExceptionStatus(Exception exception, int statusCode) {
   public void testIcebergExceptionMapper() {
     checkExceptionStatus(new IllegalArgumentException(""), 400);
     checkExceptionStatus(new ValidationException(""), 400);
-    checkExceptionStatus(new NamespaceNotEmptyException(""), 400);
+    checkExceptionStatus(new NamespaceNotEmptyException(""), 409);
```
[the remark this answers is not in the document]
Comment (lasdf1234) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestIcebergExceptionMapper.java:51]: For the Response.Status.* suggestions: this mapper and its tests intentionally mirror Iceberg's RESTCatalogAdapter / existing Gravitino tests, which use numeric status literals throughout. We only changed the mapping for NamespaceNotEmptyException (400 → 409). Refactoring the entire map/test to JAX-RS constants is out of scope for this fix.

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/IcebergExceptionMapper.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public class IcebergExceptionMapper implements ExceptionMapper<Exception> {
           .put(IllegalArgumentException.class, 400)
           .put(ValidationException.class, 400)
           .put(IllegalNameIdentifierException.class, 400)
-          .put(NamespaceNotEmptyException.class, 400)
+          .put(NamespaceNotEmptyException.class, 409)
           .put(NotAuthorizedException.class, 401)
```
[the remark this answers is not in the document]
Comment (lasdf1234) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/IcebergExceptionMapper.java:63]: To be consistent with the previous code style

**Review by roryqi — DISMISSED** [review]
LGTM.

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergNamespaceOperations.java [hunk 3]
```diff
@@ -183,6 +183,13 @@ void testDropNamespace() {
     verifyCreateNamespaceSucc(Namespace.of("drop_foo3", "a"));
     verifyDropNamespaceFail(404, Namespace.of("drop_foo3", "b"));
     verifyDropNamespaceSucc(Namespace.of("drop_foo3", "a"));
+
+    // drop non-empty namespace should return 409 Conflict per Iceberg REST spec
+    String uniqueSuffix = String.valueOf(System.nanoTime());
+    Namespace nonEmptyNs = Namespace.of("drop_nonempty_foo_" + uniqueSuffix);
+    verifyCreateNamespaceSucc(nonEmptyNs);
+    verifyRegisterTableSucc("drop_nonempty_table_" + uniqueSuffix, nonEmptyNs);
```
[the remark this answers is not in the document]
Comment (lasdf1234) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergNamespaceOperations.java]: By adopting a different processing logic, it is possible to ensure that all tables are deleted once the method is completed.

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergNamespaceOperations.java [hunk 4]
```diff
@@ -60,8 +64,12 @@
 import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
 
+@SuppressWarnings("deprecation")
```
Comment (roryqi) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergNamespaceOperations.java]: Why do we need this?

Comment (lasdf1234) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergNamespaceOperations.java]: The method NestedField.of has been marked as deprecated, so I replaced it with NestedField.required and NestedField.optional, which solved the problem and removed 'deprecated' as well.

Comment (lasdf1234) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergNamespaceOperations.java]: @SuppressWarnings("deprecation")  has been removed.

**Comment by lasdf1234** [comment]
@roryqi Thank you very much for your valuable suggestions. Does this PR have any other issues?

**Review by roryqi — APPROVED** [review]
LGTM.
